x/wasm/ioutils/ioutil.go

```diff
@@ -22,14 +21,14 @@ func Uncompress(gzipSrc []byte, limit int64) ([]byte, error) {
 	zr.Multistream(false)
 	defer zr.Close()
 	bz, err := io.ReadAll(LimitReader(zr, limit))
-	if types.ErrLimit.Is(err) {
-		return nil, errorsmod.Wrapf(err, "max %d bytes", limit)
+	if err == errLimit {
```

Comment:
Nit: my IDE complained and suggested this to cover wrapped errors, too
```suggestion
	if errors.Is(err, errLimit) {
```